ibm/mas_devops/roles/cp4d_service/tasks/determine-storage-classes.yml

```diff
@@ -14,26 +14,37 @@
   debug:
     msg: "{{ lookup_storageclasses | ibm.mas_devops.getResourceNames }}"
 
-
 # 2. Set Storage (Required)
 # -----------------------------------------------------------------------------
 - name: Default Metadata Storage if not set by user
-  when: cpd_service_storage_class is not defined or cpd_service_storage_class == ""
+  when:
+    - cpd_service_name != "wd"
+    - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
   vars:
     # ROKS, OCS, Azure
     supported_storage_classes: [ibmc-file-gold-gid, ocs-storagecluster-cephfs, azurefiles-premium]
   set_fact:
     cpd_service_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(supported_storage_classes) }}"
 
+- name: Default Metadata Storage if not set by user (Watson Discovery only)
+  when:
+    - cpd_service_name == "wd"
+    - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
+  vars:
+    # ROKS, OCS, Azure
+    supported_storage_classes: [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
+  set_fact:
+    cpd_service_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(supported_storage_classes) }}"
+
 - name: Assert that service storage class has been defined
   assert:
     that: cpd_service_storage_class is defined and cpd_service_storage_class != ""
     fail_msg: "cpd_service_storage_class must be defined"
 
-
 # 3. Debug storage class configuration
 # -----------------------------------------------------------------------------
 - name: "Debug CP4D service storage class configuration"
+  when: cpd_service_name != 'wd'
```

Comment:
Why don't we do the debug for discovery?

Reply:
fixed.